### ISSUE TITLE
Update from the Plugin "Polls" killed my whole installation

### ISSUE THREAD

**Issue by LuckySparkle** [nextcloud/polls#1385]
I wanted to update the Polls Plugin in my Nextcloud Version 20.0.6. The Nextcloud went into an error and when I now want to access my nextcloud instance the following error occurs:

Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing 'CREATE UNIQUE INDEX UNIQ_shares ON oc_polls_share (poll_id, user_id)': SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '5-Cedric' for key 'UNIQ_shares'
Detaillierte Protokollmeldungen

Update vorbereiten
Log-Level auf "debug" gesetzt
Wartungsmodus eingeschaltet
Reparaturschritt: Repair MySQL collation
Reparaturinformation: All tables already have the correct collation -> nothing to do
Reparaturschritt: Repair SQLite autoincrement
Reparaturschritt: Copy data from accounts table when migrating from ownCloud
Reparaturschritt: Drop account terms table when migrating from ownCloud
Das Datenbankschema wird aktualisiert
Datenbank aktualisiert
Reparaturschritt: Delete duplicates
Doctrine\DBAL\Exception\UniqueConstraintViolationException: An exception occurred while executing 'CREATE UNIQUE INDEX UNIQ_shares ON oc_polls_share (poll_id, user_id)': SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '5-Cedric' for key 'UNIQ_shares'

**Comment by LuckySparkle** [comment]
fixed the problem by just deleting the row in the database, the update worked now. The poll was an old one wich was allready (perma) deleted, don't know why this row was not deleted with it.

**Comment by beccon4** [comment]
I wanted to upgrade to version 1.7.5 and I got:

Es ist ein Fehler bei der Anfrage aufgetreten. Es kann nicht fortgefahren werden.
An exception occurred while executing 'CREATE UNIQUE INDEX UNIQ_options ON oc_polls_options (poll_id, poll_option_text, timestamp)': SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes

and then when I tried to recover:  (occ upgrade)

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'CREATE UNIQUE INDEX UNIQ_options ON oc_polls_options (poll_id, poll_option_text, timestamp)':

I had to disable polls to continue to work.

SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes
Update failed

My instance failed
